### ISSUE TITLE
[Menu design][Accessibility] VoiceOver focus jumps and interrupts 'Added to Shortcuts' announcement

### ISSUE THREAD

**Issue by natastro** [mozilla-mobile/firefox-ios#23488]
## Prerequisites

- Turn on VoiceOver
- Turn on treatment-a of the iOS Menu Redesign
- Found on Firefox Beta iOS app 134.0
- Open any webpage

## Steps to reproduce

1. Open Firefox Beta iOS app
2. Open the menu from the hamburger menu icon button at the bottom of the screen
3. As the action sheet appears on screen, navigate to the Save submenu and double tap to activate
4. Navigate to 'Add to Shortcuts', then double tap to activate. Observe VoiceOver focus and behavior

### Expected behavior / Actual behavior

The expectation is that the 'Added to Shortcuts' message that appears is read by VoiceOver without interruption and VoiceOver's focus will land back on the menu button in the navbar.

At this time, VoiceOver begins to read the 'Added to Shortcuts' message but is quickly interrupted by VoiceOver's focus jumping to the top on the lock icon button in the URL toolbar

### Device & build information
* Device: iPhone 15, iOS 18.1.1
* Build version: Firefox Beta 134.0
* First seen version: First time I've seen this issue
### Notes
Attachments: 

https://github.com/user-attachments/assets/4d57c9d2-978d-42d6-a369-cc687a15e6d5

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-10766)

**Comment by data-sync-user** [comment]
➤ Matt P commented:

Winnie Teichmann Andres Furlan ayeddi - Is the expected behavior mentioned by the reporter accurate? I expect it should definitely finish reading out the toast notification, but what should happen next, if anything?

Same question applies to https://mozilla-hub.atlassian.net/browse/FXIOS-10767 ( https://mozilla-hub.atlassian.net/browse/FXIOS-10767|smart-link ) and presumably saving bookmarks as well

**Comment by data-sync-user** [comment]
➤ ayeddi commented:

hi Matt P it is accurate - since the user was in the Settings / app menu, we’d expect their focus to return there. The menu is closed by now, so the hamburger menu should be the focus.

I’d note that the status update is exposed to VoiceOver (as shown on the screenshot I attach from the screen recording in the bug filed by Natalie Stroud ) - but maybe the audio of the announcement is cut off short by the default browser behavior when a page is refreshed and the focus is not explicitly managed (the VO focus gets to the top of the page/tab)

!image-20250319-155635.png|width=234,height=482,alt="image-20250319-155635.png"!

**Comment by data-sync-user** [comment]
➤ Matt P commented:

ayeddi thanks for the confirmation, didn’t even realize this was filed internally (hello Natalie Stroud 👋). Also, TIL about VO captions 😮

**Comment by data-sync-user** [comment]
➤ ayeddi commented:

Matt P correct - returning the focus to the top of the page is a default / fallback behavior on all platforms. It is not a blocker, it’s more of a papercut because when the Menu button is at the very bottom of the list, it would involve a lot of navigation gestures to get to it, especially, if the website has a ton of (properly marked up) controls.

**Comment by data-sync-user** [comment]
➤ Matt P commented:

Hey ayeddi, since there is no easy, built-in way to return focus back to the source in iOS, before charting down this path here, and presumably app-wide, I was wondering if you had a reference to the mobile WCAG for this guideline?

**Comment by data-sync-user** [comment]
➤ Matt P commented:

Note: While we determine next step with regards to where to shift focus, for now, this issue is resolved by preventing the toast notification VoiceOver announcement from being interrupted (iOS 17+).  Will create a separate ticket if we decide we need to shift focus back to the source element when a view or modal is dismissed.

**Comment by data-sync-user** [comment]
➤ Andrei Bodea commented:

Verified as fixed on v139 (55548) with iPad Air (18.2).

[^ScreenRecording_05-20-2025 17-27-57_1.MP4]
